lib/prometheus/client/data_stores/direct_file_store.rb

```diff
@@ -181,30 +192,41 @@ def process_id
           end
 
           def aggregate_values(values)
-            if @values_aggregation_mode == SUM
-              values.inject { |sum, element| sum + element }
-            elsif @values_aggregation_mode == MAX
-              values.max
-            elsif @values_aggregation_mode == MIN
-              values.min
-            elsif @values_aggregation_mode == ALL
-              values.first
+            # Each entry in the `values` array is a tuple of `value` and `timestamp`,
+            # so for all aggregations except `MOST_RECENT`, we need to only take the
+            # first value in each entry and ignore the second.
+            if @values_aggregation_mode == MOST_RECENT
+              latest_tuple = values.max { |a,b| a[1] <=> b[1] }
+              latest_tuple.first # return the value without the timestamp
             else
-              raise InvalidStoreSettingsError,
-                    "Invalid Aggregation Mode: #{ @values_aggregation_mode }"
+              values = values.map(&:first) # Discard timestamps
```

Comment:
Not sure whether we should block the PR on this, but might we be slowing down the metric read path here with extra copying?

Reply:
Possibly. Since we've done practically no optimization whatsoever on the read path, however, and we don't have any real benchmarks, i'd say let's focus on that when we work on that performance, rather than here.

I totally agree with "at leats not making it worse", but in this case, I can't think of a clear, easy alternative that still works with calling `max`, `min`, etc

Reply:
Let's focus on this when we come to optimising the read path then.